File: probability_demo/choose.cpp
#include <iostream>
#include <string>
using namespace std;

int factorial(int n) {
    if (n==0 || n==1) {
        return 1;
    }
    return n*factorial(n-1);
}

int choose(int n,int k) {
    if (k<n) { return 0; }
    return (factorial(n))/(factorial(k)*factorial(n-k));
}

int main(int argc, char* argv[]){
  int n = atoi(argv[1]);
  int k = atoi(argv[2]);
  cout<<choose(n,k)<<endl;
}

File: probability_demo/choose.py
import sys 

def factorial(n):
    if n<1:
        return 1
    return n*factorial(n-1)

def choose(n,k):
    if k<n:
        return 0
    return factorial(n)//(factorial(k)*factorial(n-k))

n = int(sys.argv[1])
k = int(sys.argv[2])

print(str(choose(n,k)))